### PR TITLE
New google api contact

### PR DESCRIPTION
**Pull request by fboitel**

### CODE REVIEW

File: htdocs/google/admin/google_contactsync.php [hunk 1]
```diff
@@ -383,14 +393,24 @@
...
+
+		$resultTags=0;
+		if (count($gContacts)) $resultTags=updateGContactGroups($gdata, $gContacts, 'contact');
+
+		if (is_numeric($resultTags) && $resultTags >= 0) {
+			$mesg .= '<br>'.$langs->trans("TagsCreatedSuccess");
+		} else {
+			$error++;
+			$errors[] = $langs->trans("Error").' '.$resultTags;
 		}
```
Comment (Hystepik) [inline htdocs/google/admin/google_contactsync.php:414]: Here you update the contact group even if the contact as not been created. Is that what you wanted to do ? And if yes why ?

File: htdocs/google/admin/google_contactsync.php [hunk 2]
```diff
@@ -443,15 +464,26 @@
...
+
+		$resultTags=0;
+		if (count($gContacts)) $resultTags=updateGContactGroups($gdata, $gContacts, 'member');
+
+		if (is_numeric($resultTags) && $resultTags >= 0) {
+			$mesg .= '<br>'.$langs->trans("TagsCreatedSuccess");
+		} else {
+			$error++;
+			$errors[] = $langs->trans("Error").' '.$resultTags;
+		}
```
Comment (Hystepik) [inline htdocs/google/admin/google_contactsync.php:485]: Here you update the contact group even if the contact as not been created. If an error occure the database and the google agenda musn't change from it's initial state.

File: htdocs/google/class/gcontacts.class.php [hunk 3]
```diff
@@ -504,7 +511,7 @@ private function fetchMemberFromDolibarr($gdata, $useremail)
 		$this->lastname = $dolContact->lastname;
 		$this->fullName = $dolContact->getFullName($langs);
 		if (empty($this->fullName)) $this->fullName=$dolContact->company;
-		$this->email = ($dolContact->email?$dolContact->email:($this->fullName.'@noemail.com'));
+		$this->email = ($dolContact->email?$dolContact->email:($this->fullName.'@noemail.com'));;
```
Comment (Hystepik) [inline htdocs/google/class/gcontacts.class.php:514]: It seams to be a forgotten semicolon

File: htdocs/google/admin/google_contactsync.php (hunk 1 above)
Comment (fboitel) [reply htdocs/google/admin/google_contactsync.php:414]: You are right, I think I can add the condition "if(!$error) {" L405

Comment (fboitel) [reply htdocs/google/admin/google_contactsync.php:414]: Another solution would be, for each contact, to update its labels right after its creation, which suggests removing the 'updateGContactGroups' function, what do you think?

File: htdocs/google/admin/google_contactsync.php (hunk 2 above)
Comment (fboitel) [reply htdocs/google/admin/google_contactsync.php:485]: https://github.com/DoliCloud/DoliMods/pull/5#discussion_r932097684

File: htdocs/google/admin/google_contactsync.php (hunk 1 above)
Comment (Hystepik) [reply htdocs/google/admin/google_contactsync.php:414]: I think the actual solution with a if(!$error) will work

**Pull request merged**